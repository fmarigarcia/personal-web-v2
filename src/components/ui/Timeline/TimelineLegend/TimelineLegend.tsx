import React from 'react';
import { useTranslation } from 'react-i18next';

export const TimelineLegend: React.FC = () => {
    const { t } = useTranslation();

    return (
        <div className="text-center text-sm text-gray-500 mt-8">
            <p className="hidden sm:block">{t('experience.hoverPreview')}</p>
            <p className="sm:hidden">{t('experience.tapSelect')}</p>
        </div>
    );
};
