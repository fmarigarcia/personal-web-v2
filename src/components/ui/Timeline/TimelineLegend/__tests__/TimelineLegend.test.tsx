import { render, screen } from '@testing-library/react';
import { TimelineLegend } from '../TimelineLegend';

// Mock react-i18next
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: jest.fn((key: string) => key),
    }),
}));

describe('TimelineLegend', () => {
    it('should render both legend texts', () => {
        render(<TimelineLegend />);

        expect(screen.getByText('experience.hoverPreview')).toBeInTheDocument();
        expect(screen.getByText('experience.tapSelect')).toBeInTheDocument();
    });

    it('should have correct responsive classes', () => {
        render(<TimelineLegend />);

        const hoverText = screen.getByText('experience.hoverPreview');
        const tapText = screen.getByText('experience.tapSelect');

        expect(hoverText).toHaveClass('hidden', 'sm:block');
        expect(tapText).toHaveClass('sm:hidden');
    });

    it('should have correct wrapper structure and classes', () => {
        const { container } = render(<TimelineLegend />);

        const wrapper = container.firstChild as HTMLElement;
        expect(wrapper).toHaveClass(
            'text-center',
            'text-sm',
            'text-gray-500',
            'mt-8'
        );
    });

    it('should translate legend keys', () => {
        render(<TimelineLegend />);

        // Verifies that t() is called with the correct keys
        expect(screen.getByText('experience.hoverPreview')).toBeInTheDocument();
        expect(screen.getByText('experience.tapSelect')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<TimelineLegend />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
