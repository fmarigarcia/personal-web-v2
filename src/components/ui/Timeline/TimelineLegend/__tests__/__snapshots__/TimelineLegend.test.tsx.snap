// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineLegend should match snapshot 1`] = `
<div
  class="text-center text-sm text-gray-500 dark:text-zinc-400 mt-8"
>
  <p
    class="hidden sm:block"
  >
    experience.hoverPreview
  </p>
  <p
    class="sm:hidden"
  >
    experience.tapSelect
  </p>
</div>
`;
